### PR TITLE
https://wso2.org/jira/browse/BPS-936

### PR DESCRIPTION
**Pull request by NatashaWso2**
Integrating the bpmn stats rest api with the existing rest api

### CODE REVIEW

File: components/bpmn/bpmn-rest/src/main/java/org/wso2/carbon/bpmn/rest/service/stats/ProcessAndTaskService.java [hunk 1]
```diff
@@ -0,0 +1,482 @@
...
+
+        for (ProcessDefinition instance : deployements) {
+            CompletedProcesses bpmnProcessInstance = new CompletedProcesses();
+            bpmnProcessInstance.setProcessDefinitionId(instance.getId());
+
+            double totalTime = 0;
+            double averageTime = 0;
+            String processDefinitionID = instance.getId();
+
+
```
Comment (jranabahu) [inline components/bpmn/bpmn-rest/src/main/java/org/wso2/carbon/bpmn/rest/service/stats/ProcessAndTaskService.java]: Remove unnecessary new lines

File: components/bpmn/bpmn-rest/src/main/java/org/wso2/carbon/bpmn/rest/service/stats/ProcessAndTaskService.java [hunk 2]
```diff
@@ -0,0 +1,482 @@
...
+     *
+     * @return array with the no. of processes started and completed over the months
+     */
+    @GET
+    @Path("/processVariation/")
+    @Produces({MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML})
+    public ResponseHolder processVariationOverTime() {
+        ResponseHolder response = new ResponseHolder();
+        List list = new ArrayList();
+        String[] MONTHS = {"Jan", "Feb", "March", "April", "May", "June", "July", "Aug", "Sep", "Oct", "Nov", "Dec"};
```
Comment (jranabahu) [inline components/bpmn/bpmn-rest/src/main/java/org/wso2/carbon/bpmn/rest/service/stats/ProcessAndTaskService.java]: Make this a constant

File: components/bpmn/bpmn-rest/src/main/java/org/wso2/carbon/bpmn/rest/service/stats/UserService.java [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+    public ResponseHolder taskVariationOverTime(@PathParam("assignee") String assignee) throws UserStoreException {
+
+        if (!(BPMNOSGIService.getUserRealm().getUserStoreManager().isExistingUser(assignee))){
+            throw new ActivitiObjectNotFoundException("Could not find user with id '" +
+                    assignee + "'.");
+        }
+
+        ResponseHolder response = new ResponseHolder();
+        List list = new ArrayList();
+        String[] MONTHS = {"Jan", "Feb", "March", "April", "May", "June", "July", "Aug", "Sep", "Oct", "Nov", "Dec"};
```
Comment (jranabahu) [inline components/bpmn/bpmn-rest/src/main/java/org/wso2/carbon/bpmn/rest/service/stats/UserService.java]: Make this a constant

File: components/bpmn/bpmn-rest/src/main/java/org/wso2/carbon/bpmn/rest/service/stats/ProcessAndTaskService.java (hunk 2 above)
Comment (NatashaWso2) [inline components/bpmn/bpmn-rest/src/main/java/org/wso2/carbon/bpmn/rest/service/stats/ProcessAndTaskService.java]: Already fixed 
https://github.com/wso2/carbon-business-process/blob/master/components/bpmn/org.wso2.carbon.bpmn.rest/src/main/java/org/wso2/carbon/bpmn/rest/service/stats/ProcessAndTaskService.java#L57

File: components/bpmn/bpmn-rest/src/main/java/org/wso2/carbon/bpmn/rest/service/stats/UserService.java (hunk 3 above)
Comment (NatashaWso2) [inline components/bpmn/bpmn-rest/src/main/java/org/wso2/carbon/bpmn/rest/service/stats/UserService.java]: Already fixed 
https://github.com/wso2/carbon-business-process/blob/master/components/bpmn/org.wso2.carbon.bpmn.rest/src/main/java/org/wso2/carbon/bpmn/rest/service/stats/ProcessAndTaskService.java#L57

File: components/bpmn/bpmn-rest/src/main/java/org/wso2/carbon/bpmn/rest/service/stats/ProcessAndTaskService.java (hunk 1 above)
Comment (NatashaWso2) [inline components/bpmn/bpmn-rest/src/main/java/org/wso2/carbon/bpmn/rest/service/stats/ProcessAndTaskService.java]: Fixed with commit 226ec4f560fcbb6964e0ad72e9b915e2509b62af

File: components/bpmn/bpmn-rest/src/main/java/org/wso2/carbon/bpmn/rest/service/stats/UserService.java (hunk 3 above)
Comment (NatashaWso2) [inline components/bpmn/bpmn-rest/src/main/java/org/wso2/carbon/bpmn/rest/service/stats/UserService.java]: Fixed with commit 226ec4f560fcbb6964e0ad72e9b915e2509b62af

**Pull request closed without merging**